src/pepr/operator/utils/waypoint.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Waypoint utility module for standardized waypoint operations
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { UDSPackage } from "../crd";
+import { Mode } from "../crd/generated/package-v1alpha1";
+// No logger needed in this utility module
+
+// Constants for waypoint configuration
+const WAYPOINT_SUFFIX = "-waypoint"; // Suffix for waypoint resource names
+const WAYPOINT_PREFIX = "uds-core-"; // Prefix for waypoint resource names
+
+// Cache for waypoint names to avoid regenerating them
+const waypointNameCache = new Map<string, string>();
+
+/**
+ * Determines if a package should use ambient waypoint networking
+ * @param pkg - The UDS package to check
+ * @returns boolean indicating if ambient waypoint should be used
+ */
+export const shouldUseAmbientWaypoint = (pkg?: UDSPackage): boolean => {
+  if (!pkg) return false;
+
+  // Check if package has ambient mode and authservice SSO
```

Comment:
Making `pkg` an optional param here feels odd - if it were required you could skip this check?
```suggestion
export const shouldUseAmbientWaypoint = (pkg: UDSPackage): boolean => {
  // Check if package has ambient mode and authservice SSO
```